src/absences/absences.test.ts

```diff
@@ -1,32 +1,35 @@
 import Absences from './index';
-import { Absence } from './types';
+import { Absence, AbsenceApprove, AbsenceCancel, AbsenceReject, AbsenceUpdate } from './types';
 import axios from 'axios';
 import AxiosMockAdapter from 'axios-mock-adapter';
-import RequestParams from '../utils/requestParams/requestParams';
+import { successfulResponse } from '../../test/utils/successfulResponse';
 
 describe('Absences', () => {
-  var absences: Absences = new Absences({});
-  var readPath: string = `${absences.getResourcePath()}/read`;
-  var mock = new AxiosMockAdapter(axios);
-  var result: Promise<Absence[]> | null;
-  var resultSingle: Promise<Absence> | null;
+  const absences: Absences = new Absences({});
+
+  const readPath: string = `${absences.getResourcePath()}/read`;
+  const createPath: string = `${absences.getResourcePath()}/create`;
+  const updatePath: string = `${absences.getResourcePath()}/update`;
+  const deletePath: string = `${absences.getResourcePath()}/delete`;
+  const approvePath: string = `${absences.getResourcePath()}/approve`;
+  const rejectPath: string = `${absences.getResourcePath()}/reject`;
+  const cancelPath: string = `${absences.getResourcePath()}/cancel`;
+
+  const mock = new AxiosMockAdapter(axios);
 
   afterEach(() => {
     mock.reset();
-    result = null;
-    resultSingle = null;
   });
 
   test('read', async () => {
     mock.onGet(readPath).reply(200, { Success: true, NumResults: 1, Results: [{}] });
-    result = absences.read();
-    await result.then((result) => expect(result).toStrictEqual([{}]));
+    const result: Absence[] = await absences.read();
```

Comment:
What is the reason for this change?

Reply:
Removing `vars` that need to be reset in `afterEach`. I believe they were added to verify that the type of eg. `absences.read()` is `Promise<Absence[]>`, so I made sure to keep that type verification by explicitly specifying the type on the result variable inside of each test. Then we also don't need different variables like `result` and `resultSingle`. I generally like to replace `then` with `await` to improve readability by removing unnecessary arrow functions.